src/render/OpenGL.cpp

```diff
@@ -1083,7 +1083,7 @@ void CHyprOpenGLImpl::renderRoundedShadow(wlr_box* box, int round, int range, fl
     RASSERT((box->width > 0 && box->height > 0), "Tried to render shadow with width/height < 0!");
     RASSERT(m_pCurrentWindow, "Tried to render shadow without a window!");
 
-    if (!pixman_region32_not_empty(m_RenderData.pDamage))
+    if (!pixman_region32_not_empty(m_RenderData.pDamage) || (m_pCurrentWindow && m_pCurrentWindow->m_sAdditionalConfigData.forceNoShadow))
```

Comment:
it's better to nope earlier - in the decoration class itself. see CHyprDropShadowDecoration::render

Reply:
There's no `::render` that I can find, but there's `::draw`. Should I go for draw?

Reply:
yeah I was paraphrasing